improver/blending/calculate_weights_and_blend.py

```diff
@@ -256,17 +262,17 @@ def process(
         cube = merger(cubelist, cycletime=cycletime)
 
         if "model" in self.blend_coord:
-            self.blend_coord = "model_id"
-
-        coord_names = [coord.name() for coord in cube.coords()]
-        if (
-            self.blend_coord not in coord_names
-            or len(cube.coord(self.blend_coord).points) == 1
-        ):
-            result = self._update_metadata_only(cube, attributes_dict, cycletime)
+            self.blend_coord = copy(MODEL_BLEND_COORD)
+
+        coords_to_remove = get_coords_to_remove(cube, self.blend_coord)
+
+        weights = self._calculate_blending_weights(cube)
+        cube, weights = self._remove_zero_weighted_slices(cube, weights)
```

Comment:
(See above comment regarding editing in place, if extract can be done in place, the cube and weights wouldn't need to be reassigned)